commit ecaa02371ef6bb9a1f79f5dc0f2ab3cec3793332 (refs/changes/36/936/1)
Author: Cody Peter Mello <cody.mello@joyent.com>
Date:   2016-11-21T22:43:41+00:00 (2 years, 11 months ago)
    
    MORAY-378 Bucket cache isn't cleared on bucket update or delete

diff --git a/lib/buckets/common.js b/lib/buckets/common.js
index 1990dbb..fdc8d75 100644
--- a/lib/buckets/common.js
+++ b/lib/buckets/common.js
@@ -14,6 +14,8 @@ var assert = require('assert-plus');
 var once = require('once');
 var vasync = require('vasync');
 
+var objCommon = require('../objects/common');
+
 var mod_errors = require('../errors');
 var InvalidBucketConfigError = mod_errors.InvalidBucketConfigError;
 var InvalidBucketNameError = mod_errors.InvalidBucketNameError;
@@ -243,6 +245,12 @@ function validateBucket(req, cb) {
 }
 
 
+function shootdownBucket(req, cb) {
+    objCommon.shootdownBucket(req);
+    cb();
+}
+
+
 
 ///--- Exports
 
@@ -253,5 +261,6 @@ module.exports = {
     buildIndexString: buildIndexString,
     createIndexes: createIndexes,
     mapIndexType: mapIndexType,
+    shootdownBucket: shootdownBucket,
     validateBucket: validateBucket
 };
diff --git a/lib/buckets/del.js b/lib/buckets/del.js
index 85fc8ad..59375e2 100644
--- a/lib/buckets/del.js
+++ b/lib/buckets/del.js
@@ -193,7 +193,8 @@ function del(options) {
                 deleteConfig,
                 dropTable,
                 dropSequence,
-                dropLockingSerial
+                dropLockingSerial,
+                common.shootdownBucket
             ]
         });
     }
diff --git a/lib/buckets/update.js b/lib/buckets/update.js
index b7ad783..25c5985 100644
--- a/lib/buckets/update.js
+++ b/lib/buckets/update.js
@@ -17,7 +17,6 @@ var clone = require('clone');
 
 var common = require('./common');
 var control = require('../control');
-var objCommon = require('../objects/common');
 
 var mod_errors = require('../errors');
 var BucketNotFoundError = mod_errors.BucketNotFoundError;
@@ -484,7 +483,8 @@ function update(options) {
                 dropColumns,
                 addColumns,
                 createIndexes,
-                createUniqueIndexes
+                createUniqueIndexes,
+                common.shootdownBucket
             ]
         });
     }
